### PR TITLE
Add support for filtering messages out before dumping to storage

### PR DESCRIPTION
**Pull request by Danm87**
implementing #399

### CODE REVIEW

**Comment by Danm87** [comment]
implementing #399

File: src/main/java/com/pinterest/secor/consumer/Consumer.java [hunk 1]
```diff
@@ -139,9 +139,14 @@ protected boolean consumeNextMessage() {
             } catch (IOException e) {
                 throw new RuntimeException("Failed to adjust offset.", e);
             }
+
+            Message transformedMessage = mMessageTransformer.transform(rawMessage);
```
Comment (HenryCaiHaiying) [inline src/main/java/com/pinterest/secor/consumer/Consumer.java]: Why the transform call is moved outside try/catch block?

**Comment by Danm87** [comment]
by mistake, 
returned it back where it belonged :)

**Comment by HenryCaiHaiying** [comment]
Thanks for the contribution.

**Pull request merged**